app/grants/views.py

```diff
@@ -258,10 +260,16 @@ def grant_details(request, grant_id, grant_slug):
             team_members = request.POST.getlist('edit-grant_members[]')
             team_members.append(str(grant.admin_profile.id))
             grant.team_members.set(team_members)
+
+            '''deleting all is just to support a single category from a UI point of view'''
+            grant.categories.all().delete()
```

Comment:
why do we do this ?

Reply:
As a way of overwriting what’s there

Reply:
Doesn't this delete the queryset of grant.categories? There's also an on_delete=cascade option on the grant field in the GrantCategory model so it will delete the grant as well. I think this needs re-architecting. You should add a ManyToMany field on the grant model itself that points to the GrantCategory object, and then you can easily use `clear()` or `remove()` to manage the changes - or in this case we can probably just set them directly. https://docs.djangoproject.com/en/3.0/topics/db/examples/many_to_many/

Reply:
That's a fair point thanks @danlipert I'll dive into this